packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -444,6 +467,30 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
      * when the op is ACKed or resubmitted, respectively
      */
     protected abstract applyStashedOp(content: any): unknown;
+
+    public emit(
+        event: EventEmitterEventType,
+        ...args: any[]): boolean {
+        let returnValue: boolean;
+
+        // Do not include the events emmited by SharedObject itself, we only care
+        // about the events from child classes.
+        if (["op", "pre-op"].includes(event.toString())) {
+            returnValue = super.emit(event, ...args);
+        } else {
+            returnValue = ExecuteWithAggregatedTelemetry(
```

Comment:
i would also like to see this class flush its current data on dispose

Reply:
if we go with an object based approach the object could also be disposable, and flush to the log when it is disposed to encapsulate it